outlines/processors/base_logits_processor.py

```diff
@@ -81,6 +81,15 @@ def __call__(
         torch_logits = self._to_torch(logits)
         input_ids = self._to_torch(input_ids)
 
+        # if input_ids is 1D and logits is 2D with a single sequence,
```

Comment:
Not for this PR, but I think we'll have to break down the logits processor into library-specific ones. On top of this we'll soon have methods in `outlines-core` to mask on-device that see optimized for `torch`, MLX and `NumPy`. We can already open an issue for that.